server/server.go

```diff
@@ -9,17 +9,67 @@ import (
 
 	"github.com/bombsimon/amqp-rpc/connection"
 	"github.com/bombsimon/amqp-rpc/logger"
-	"github.com/bombsimon/amqp-rpc/middleware"
+)
+
+type ctxKey string
+
+const (
+	// CtxQueueName can be used to get the queue name from the context.Context
+	// inside the HandlerFunc.
+	CtxQueueName ctxKey = "queue_name"
```

Comment:
Is it possible to treat this like the enum simulation in Go using integers and iota?

```go
const (
    CtxQueueName int = iota
    CtxUserId
    CtxUUID
)
```

Reply:
As long as the `CtxXXX` is it's own type `iota` can be used, yes.